Configuration.md
# Introduction #

This document is meant to ease the installation of assetr. Make sure to hop on the Google Groups for assetr if you have any questions.

## Required Settings ##
### SQL Settings ###

The purpose for calling the array variable 'sql' is to allow for portability to the PostgreSQL version.

```
$config['sql']['server'] = "localhost";
```
  * This specifies the SQL server to connect to.

```
$config['sql']['user'] = "root";
```
  * This specifies the SQL user to use.

```
$config['sql']['password'] = "somepw123!";
```
  * This specifies the SQL user's password to the SQL server.

```
$config['sql']['database'] = "assetr";
```
  * This is the database that the assetr tables will be housed. Make sure to use a valid database.

```
$config['sql']['pre'] = "";
```
  * This is the prefix for the assetr tables. This is useful if you have multiple applications sharing a single database, or your webhost only gives you one database. Either way, if you specify a prefix, then you need to edit the .sql file and change the names of the tables to reflect the prefix.

### Cookie Settings ###
These are to help you specify settings dealing with the use of cookies.

```
$config['cookie']['keeptime'] = time()+(60 * 60 * 24 * 90);
```
  * This sets the cookie's alive time. Once the cookie hits the time (above is 90 days), the cookie is destroyed. The method above is ideal for logging the cookie's alive time. If you wish to change the length of time the cookie is alive for, then change the information inside the parenthesis (seconds x minutes x hours x days[x weeks[ x years ](.md)]) etc. To kill the auto-login just make the variable blank.

```
$config['cookie']['domain'] = "domainname.tld";
```
  * This is to specify the IP address or domain that the cookies for assetr are linked to.

### File and Folder Settings ###
These are specific to the file/folder creation and saving functions in assetr.

```
$config['folder']['absolute'] = "/var/www/assetr"; 
```
  * The absolute folder path to assetr (preferably where index.php and config.php are housed) on the server. This cannot include a trailing slash, or assetr will break.

```
$config['folder']['rep'] = "/repositories";
```
  * This specifies where the repositories folder is located in the assetr directory. This is not supposed to be absolute path on the server. This cannot include a trailing slash or assetr will break.

```
$config['folder']['permission'] = "0777";
```
  * For all new folders, they need to be created with a permission. This is where you can specify what the permission settings will be.

```
$config['file']['permission'] = "0777";
```
  * For all new files, they need to be created with a permission. This is where you can specify what the permission settings will be.

## Optional Settings ##
### SMTP Settings ###
assetr has a built in ability to e-mail files to a user checking out, as well as inform others of the change.

### IMAP Settings ###
assetr has a built in ability to recieve e-mailed files for input in to a repository.

0.1/config.php
<?php
	/**
	 * assetr v0.1 <http://code.google.com/p/assetr/>
	 *
	 * Copyright 2008 Duane Jeffers
	 * This file is part of assetr.
	 *
	 * assetr is free software: you can redistribute it and/or modify
	 * it under the terms of the GNU General Public License as published by
	 * the Free Software Foundation, either version 3 of the License, or
	 * (at your option) any later version.
	 *
	 * assetr is distributed in the hope that it will be useful,
	 * but WITHOUT ANY WARRANTY; without even the implied warranty of
	 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	 * GNU General Public License for more details.
	 *
	 * You should have received a copy of the GNU General Public License
	 * along with assetr.  If not, see <http://www.gnu.org/licenses/>.
	 *
	 * What is this file?:
	 * config.php - This is the configuration file for assetr. - Need help? Check out: http://code.google.com/p/assetr/wiki/Configuration
	 **/

// SQL info
$config['sql']['server'] = "localhost"; // Server location
$config['sql']['user'] = ""; // Username
$config['sql']['password'] = ""; // Password
$config['sql']['database'] = "assetr"; // The Database
$config['sql']['pre'] = ""; // Any prefixes?
$config['sql']['driver'] = "mysql"; // This is the type of database server. Make sure you have the appropriate driver for the type of DB.

// Cookie Variables
// Login Time. This allows for manual override of the cookie's timeout.
$config['cookie']['keeptime'] = time()+(60 * 60 * 24 * 90); // This will make the cookie timeout after 90 days. 
//Adversely, you can use mktime to set a time in the future, but this makes it timeout at a specific day, not based on when the login occurred.
//$config['cookie']['keeptime'] = mktime(0, 0, 0, 13, 2, 2010); // This will make the cookie timeout on February 13, 2010 at 00:00 (Midnight).
$config['cookie']['domain'] = "domainname.tld"; // You need to set this so you can specify what domain the cookie can only be accessed on.

// Folder Settings. - These are required for normal use.
$config['folder']['absolute'] = "/var/www/assetr"; // This is the absolute path where assetr is located on the server. (And please leave off the trailing slash).
$config['folder']['rep'] = "/repositories"; // This is the folder that holds the repositories. It should be located in the assetr directory.
$config['folder']['permission'] = "0777"; // Change the new folder permission settings. This is required for use.

// File Settings - These are required for normal use.
$config['file']['permission'] = "0777"; // This will set all the new files to 777.

// SMTP Settings - this is not required for normal use.
$config['smtp']['server'] = "localhost";
$config['smtp']['user'] = "username";
$config['smtp']['password'] = "password";
$config['smtp']['fromemail'] = "Name <email@localhost>";

// IMAP Settings - this is not required for normal use.
$config['imap']['server'] = "localhost";
$config['imap']['user'] = "username";
$config['imap']['password'] = "password";
$config['imap']['emailaddress'] = "email@localhost";

?>
